applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h

```diff
@@ -0,0 +1,405 @@
+
+    /**
+     * @brief Finalize the material response in terms of 2nd Piola-Kirchhoff stresses
+     * @see Parameters
+     */
+    void FinalizeMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues) override;
+
+    /**
+     * @brief Finalize the material response in terms of Kirchhoff stresses
+     * @see Parameters
+     */
+    void FinalizeMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues) override;
+    ///@}
+    ///@name Access
+    ///@{
+
+    /**
+     * @brief This is to be called at the very beginning of the calculation
+     * @details (e.g. from InitializeElement) in order to initialize all relevant attributes of the constitutive law
+     * @param rMaterialProperties the Properties instance of the current element
+     * @param rElementGeometry the geometry of the current element
+     * @param rShapeFunctionsValues the shape functions values in the current integration point
+     */
+    void InitializeMaterial(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const Vector& rShapeFunctionsValues
+        ) override;
+
+    double GetFatigueReductionFactor() {return mFatigueReductionFactor;}
+    void SetFatigueReductionFactor(const double toFred) {mFatigueReductionFactor = toFred;}
+
+    Vector GetPreviousStresses() {return mPreviousStresses;}
+    void SetPreviousStresses(const Vector toPreviousStresses) {mPreviousStresses = toPreviousStresses;}
+
+    double GetMaxStress() {return mMaxStress;}
+    void SetMaxStress(const double toMaxStress) {mMaxStress = toMaxStress;}
+
+    double GetMinStress() {return mMinStress;}
+    void SetMinStress(const double toMinStress) {mMinStress = toMinStress;}
+
+    double GetPreviousMaxStress() {return mPreviousMaxStress;}
+    void SetPreviousMaxStress(const double toPreviousMaxStress) {mPreviousMaxStress = toPreviousMaxStress;}
+
+    double GetPreviousMinStress() {return mPreviousMinStress;}
+    void SetPreviousMinStress(const double toPreviousMinStress) {mPreviousMinStress = toPreviousMinStress;}
+
+    unsigned int GetNumberOfCycles() {return mNumberOfCycles;}
+    void AddCycle() {mNumberOfCycles++;}
+    void SetNumberOfCycles(const unsigned int toCycles) {mNumberOfCycles = toCycles;}
+
+    double GetReversionFactor() {return mReversionFactor;}
+    void SetReversionFactor(const double toReversionFactor) {mReversionFactor = toReversionFactor;}
+
+    double GetFatigueReductionParameter() {return mFatigueReductionParameter;}
+    void SetFatigueReductionParameter(const double toFatigueReductionParameter) {mFatigueReductionParameter = toFatigueReductionParameter;}
+
+    Vector GetStressVector() {return mStressVector;}
+    void SetStressVector(const Vector toStressVector) {mStressVector = toStressVector;}
+
+    void ResetCycleCounter(){mHasCountedCycle = false;}
+    void SetCycleCounter(const bool tocycle){mHasCountedCycle = tocycle;}
+    bool GetCycleCounter() {return mHasCountedCycle;}
```

Comment:
I am not sure if we should extend the public interface of a CLaw

@RiccardoRossi what do you say here?